pages/registration.js

```diff
@@ -0,0 +1,78 @@
+import { useState } from 'react';
+import Link from 'next/link';
+import { useRouter } from 'next/router';
+
+import { Layout } from '../components/Layout';
+import { useAuth } from '../contexts/AuthContext';
+import { FormInput } from '../components/Form/FormInput';
+
+const Registration = () => {
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
+  const { signup } = useAuth();
+  const router = useRouter();
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setError('');
+    try {
```

Comment:
The same new lines are mentioned in the comments above. In general, make sure you separate blocks of code by new lines for better readability.

It separates it visually and mentally and just strives more towards good code.